### ISSUE TITLE
Switch to Qt 5

### ISSUE THREAD

**Issue by stephanlachnit** [benjamimgois/goverlay#72]
Currently GOverlay still uses Gtk 2, however this might be a problem for a potential Debian (and Ubuntu) package, see [#947713](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=947713#5).

Building with Gtk 3 can be done by adding "--ws=gtk3" to the build options, however it looks awful:
![Screenshot from 2020-11-04 15-34-18](https://user-images.githubusercontent.com/36662302/98124735-debf7280-1eb3-11eb-9be1-e71083396f11.png)
The compilation and execution also give some warnings.

Can you try to give it a shot? Lazarus on Debian is now also compiled with Gtk 3, so it should be possible somehow (not saying it's easy).

**Comment by benjamimgois** [comment]
@stephanlachnit i've got similar problems with QT5 on Lazarus. Never tried the GTK3 option, since they say it still in early alpha support on lazarus. I'll take a look at a SVN lazarus build.

**Comment by stephanlachnit** [comment]
Tbh Qt5 looks way more usable
![Screenshot from 2020-11-04 15-55-25](https://user-images.githubusercontent.com/36662302/98126940-6dcd8a00-1eb6-11eb-879e-9403400efefa.png)

But I think it doesn't really fit the style of GOverlay.

**Comment by benjamimgois** [comment]
I think qt5 will be the way to go. GTK3 seems abandoned in lazarus. I got to fix some transparency issues with the QT5 version... Is actually looking good, the problem reside in some QT themes that doesn't render some elements like group-boxes frames. Manjaro uses the Kvantum engine  + Matchama theme to render QT applications. I don't know if it's a standard or if other distros use diferent QT engines.

![goverlay_qt5](https://user-images.githubusercontent.com/7688496/98270669-daf02680-1f6d-11eb-80d3-37a2d14cd761.png)


 Do you know what qt engine debian uses @stephanlachnit ?

**Comment by stephanlachnit** [comment]
> I got to fix some transparency issues with the QT5 version... Is actually looking good, the problem reside in some QT themes that doesn't render some elements like group-boxes frames. 

That looks pretty good!

>  Do you know what qt engine debian uses @stephanlachnit ?

No I don't, but if you share the sources (for example on a new branch), I send a screenshot on how it looks on Debian w/ Gnome.

**Comment by benjamimgois** [comment]
Maybe it would be possible to force default qt theme for goverlay, so we have a homogeneous look for the app across distros in qt5. QT5 also fixes the problem with HDPI displays

**Comment by benjamimgois** [comment]
> > Do you know what qt engine debian uses @stephanlachnit ?
> 
> No I don't, but if you share the sources (for example on a new branch), I send a screenshot on how it looks on Debian w/ Gnome.

Just created a branch "qt5" with the fixes. Also discovered that if you compile the app in qt5 you can force a diferent theme with the option `-style`

![image](https://user-images.githubusercontent.com/7688496/98279938-5f947200-1f79-11eb-9bd9-d37b01b7d5b6.png)

**Comment by yochananmarqos** [comment]
Looks good with Adwaita-dark on Manjaro GNOME using QGnomePlatform. I'm not using Kvantum.

![Screenshot from 2020-11-05 11-21-35](https://user-images.githubusercontent.com/5437803/98280872-44b20580-1f59-11eb-80fe-8a8d829bc11b.png)

Why does it still need GTK2?

```
goverlay-git I: Link-level dependence (atk) in file ['usr/lib/libatk-1.0.so.0']
goverlay-git I: Link-level dependence (cairo) in file ['usr/lib/libcairo.so.2']
goverlay-git I: Link-level dependence (gdk-pixbuf2) in file ['usr/lib/libgdk_pixbuf-2.0.so.0']
goverlay-git I: Link-level dependence (glib2) in file ['usr/lib/libgobject-2.0.so.0', 'usr/lib/libglib-2.0.so.0', 'usr/lib/libgmodule-2.0.so.0', 'usr/lib/libgthread-2.0.so.0']
goverlay-git I: Link-level dependence (glibc) in file ['usr/lib/libc.so.6', 'usr/lib/libpthread.so.0', 'usr/lib/libdl.so.2']
goverlay-git I: Link-level dependence (gtk2) in file ['usr/lib/libgdk-x11-2.0.so.0', 'usr/lib/libgtk-x11-2.0.so.0']
goverlay-git I: Link-level dependence (libx11) in file ['usr/lib/libX11.so.6']
goverlay-git I: Link-level dependence (pango) in file ['usr/lib/libpango-1.0.so.0']
goverlay-git W: Unused shared library '/usr/lib/libgmodule-2.0.so.0' by file ('usr/bin/goverlay')
goverlay-git W: Unused shared library '/usr/lib/libcairo.so.2' by file ('usr/bin/goverlay')
goverlay-git W: Unused shared library '/usr/lib/libatk-1.0.so.0' by file ('usr/bin/goverlay')
goverlay-git E: Dependency gtk2 detected and not included (libraries ['usr/lib/libgdk-x11-2.0.so.0', 'usr/lib/libgtk-x11-2.0.so.0'] needed in files ['usr/bin/goverlay'])
```

**Comment by benjamimgois** [comment]
@yochananmarqos that looks great ! How do i activate qgnomeplatform instead kvantum ?

**Comment by yochananmarqos** [comment]
If you're using Manjaro GNOME, QGnomePlatform should already be installed. Technically nothing needs to be done as it's used automatically on GTK-based DE's when installed. However, Manjaro does set the `QT_QPA_PLATFORMTHEME` environment variable to `gnome`. You can check by running:

```
echo $QT_QPA_PLATFORMTHEME
```

It should return `gnome` or no output.

To disable using Kvantum, comment out this line in `/etc/environment`. You may need to logout / reboot for it to take effect:

```
QT_STYLE_OVERRIDE=kvantum
```

**Comment by benjamimgois** [comment]
@yochananmarqos i follow your instructions but the interface is rendering very diferent then yours. Take a look :+1: 

![image](https://user-images.githubusercontent.com/7688496/98285109-d41edf00-1f80-11eb-80aa-663b455beac9.png)

Did you compile the app forcing the qt5 toolkit ? otherwise it will fallback to gtk2.

`lazbuild -B goverlay.lpi --ws=qt5`

**Comment by yochananmarqos** [comment]
> the interface is rendering very diferent then yours.

What theme did you choose in GNOME Tweaks? Looks like Matcha-dark-something. If you changed it to Adwaita-dark, then it appears it's still using a Kvantum theme. What's the output of:

```
echo $QT_STYLE_OVERRIDE
```

> Did you compile the app forcing the qt5 toolkit?

Whoops. Now I did.

![Screenshot from 2020-11-05 12-14-04](https://user-images.githubusercontent.com/5437803/98285987-8d20f180-1f60-11eb-86d6-03aa3af76593.png)

**Comment by stephanlachnit** [comment]
Still has white UI elements for me:
![Screenshot from 2020-11-05 20-24-17](https://user-images.githubusercontent.com/36662302/98286963-00e4ed80-1fa5-11eb-8a39-2be0662cdc58.png)

I use adwaita-dark

**Comment by benjamimgois** [comment]
> What theme did you choose in GNOME Tweaks? 

Matcha-dark-sea

> ```
> echo $QT_STYLE_OVERRIDE
> ```
kvantum


> 
> Also check the contents of the following files if they exist:
> ```
> ~/.pam_environment
> ~/.config/environment.d/envvars.conf
> ```
> 
None of then exist in my system.

I think i've got the best results so far forcing the "fusion" style

![image](https://user-images.githubusercontent.com/7688496/98287645-64125800-1f84-11eb-81ee-485c01e84516.png)

**Comment by stephanlachnit** [comment]
Ahh now I got it. I didn't have any styles installed, so it used "Fusion" by default. After installing Adwaita, I get this:

Adwaita dark (looks ok-ish)
![Screenshot from 2020-11-05 20-33-10](https://user-images.githubusercontent.com/36662302/98287840-40600980-1fa6-11eb-8585-bf3c6c71eddd.png)

Adwaita (looks terrible)
![Screenshot from 2020-11-05 20-34-19](https://user-images.githubusercontent.com/36662302/98287973-6c7b8a80-1fa6-11eb-94d7-e68b0ee5913f.png)

However, since I guess that "Fusion" will be the default for most Gtk based setups, it's probably a good idea to make that theme look good.

**Comment by benjamimgois** [comment]
> However, since I guess that "Fusion" will be the default for most Gtk based setups, it's probably a good idea to make that theme look good. Edit: why is your Fusion dark but not mine :c

If i use matcha-dark-see on gnome-tweaks all the elements become dark, if i use the light matcha-sea i got the same results as yours.

**Comment by benjamimgois** [comment]
QT5 implemented in 0.4.1
